### PR TITLE
Enhancement optimize image scaledown

### LINKED ISSUE #4343: Important performance drop when using medium texture quality
**Describe the Bug**

Set your OpenGL settings to:
- occlusion: disabled
- texture quality: **high**
- filtering: 4
- shadows: unchecked
- AA: unchecked

run:
`./webots projects/robots/dji/mavic/worlds/mavic_2_pro.wbt`
it takes about 10 seconds to load.

Now change the texture quality from  **high** to **medium**, close webots and run the same command again.
Now it takes ~44 seconds to load.

The effect varies from world to world, but the mavic is one particularly noticeable. What about it is the issue I'm not sure. either way, it doesn't seem a normal behavior when lowering the settings.

### PR DESCRIPTION
**Pull request by BenjaminDeleze**
Fix #4343 

Tasks:

- [x] Replace downscale functions by Qt ones
- [x] Add a preference option to select (Qt smooth) or not (Qt Fast) the blur-

Here are the performance test I made (time is in seconds):

In debug mode:

|            | **Qt fast** | **Qt smooth** | **Downscale without Gaussian** | **Downscale with Gaussian** | **Downscale with Gaussian and openmp** |
|------------|-------------|---------------|--------------------------------|-----------------------------|----------------------------------------|
| **Low**    | 0.03        | 0.12          | 0.49                           | 4.1                         | 0.5                                    |
| **Medium** | 0.1         | 0.13          | 1.9                            | 16.2                        | 1.6                                    |

In release mode:

|            | **Qt fast** | **Qt smooth** | **Downscale without Gaussian** | **Downscale with Gaussian** | **Downscale with Gaussian and openmp** |
|------------|-------------|---------------|--------------------------------|-----------------------------|----------------------------------------|
| **Low**    | 0.03        | 0.12          | 0.06                           | 0.43                        | 0.11                                   |
| **Medium** | 0.1         | 0.13          | 0.26                           | 1.78                        | 0.38                                   |

And here are the different results:

High texture quality:
high
Low texture quality:
low
Low texture quality with Gaussian blur:
low_with_gaussian_blue
Low texture quality with Qt fast:
qt_fast
Low texture with Qt smooth:
qt_smooth

You can remark that Qt smooth is a bit less blurry than the Gaussian blur.

There was also an old comment saying that Qt smooth was altering the alpha channel but look at several worlds and saw nothing strange. So for me it seems ok.

### CODE REVIEW

File: src/webots/maths/WbImage.hpp [hunk 1]
```diff
@@ -24,8 +24,6 @@ class WbImage {
   int width() const { return mWidth; }
   int height() const { return mHeight; }
 
-  WbImage *downscale(int width, int height, int xBlurRadius, int yBlurRadius);
-
```
Comment (omichel) [inline src/webots/maths/WbImage.hpp:28]: Can't we get rid of that class? It seems pretty useless now.

Comment (BenjaminDeleze) [reply src/webots/maths/WbImage.hpp:28]: Indeed

**Review by omichel — CHANGES_REQUESTED** [review]
I would also add a line in the change log, as it is a good improvement over R2022a.

**Review by omichel — APPROVED** [review]
Excellent.

**Pull request merged**